rest/src/main/java/com/yahoo/ycsb/webservice/rest/RestClient.java

```diff
@@ -0,0 +1,367 @@
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("GET Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status insert(String table, String endpoint, HashMap<String, ByteIterator> values) {
+    int responseCode;
+    try {
+      responseCode = httpExecute(new HttpPost(urlPrefix + endpoint), values.get("data").toString());
+    } catch (Exception e) {
+      responseCode = handleExceptions(e, urlPrefix + endpoint, HttpMethod.POST);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("POST Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status delete(String table, String endpoint) {
+    int responseCode;
+    try {
+      responseCode = httpDelete(urlPrefix + endpoint);
+    } catch (Exception e) {
+      responseCode = handleExceptions(e, urlPrefix + endpoint, HttpMethod.DELETE);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("DELETE Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status update(String table, String endpoint, HashMap<String, ByteIterator> values) {
+    int responseCode;
+    try {
+      responseCode = httpExecute(new HttpPut(urlPrefix + endpoint), values.get("data").toString());
+    } catch (Exception e) {
+      responseCode = handleExceptions(e, urlPrefix + endpoint, HttpMethod.PUT);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("PUT Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status scan(String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    return Status.NOT_IMPLEMENTED;
+  }
+
+  // Maps HTTP status codes to YCSB status codes.
+  private Status getStatus(int responseCode) {
+    int rc = responseCode / 100;
+    if (responseCode == 400) {
+      return Status.BAD_REQUEST;
+    } else if (responseCode == 403) {
+      return Status.FORBIDDEN;
+    } else if (responseCode == 404) {
+      return Status.NOT_FOUND;
+    } else if (responseCode == 501) {
+      return Status.NOT_IMPLEMENTED;
+    } else if (responseCode == 503) {
+      return Status.SERVICE_UNAVAILABLE;
+    } else if (rc == 5) {
+      return Status.ERROR;
+    }
+    return Status.OK;
+  }
+
+  private int handleExceptions(Exception e, String url, String method) {
+    if (logEnabled) {
+      System.out.println(new StringBuilder(method).append(" Request: ").append(url).append(" | ")

```

Comment:
The `System.out` in a few places should be `System.err` to make sure that messages go to stderr. I think there are a few examples of this in other bindings.